java-extensions/paimon-reader/src/main/java/com/starrocks/paimon/reader/PaimonSplitScanner.java

```diff
@@ -0,0 +1,214 @@
+        this.tableName = params.get("table_name");
+        this.requiredFields = params.get("required_fields").split(",");
+        this.splitInfo = params.get("split_info");
+        this.classLoader = this.getClass().getClassLoader();
+        for (Map.Entry<String, String> kv : params.entrySet()) {
+            LOG.debug("key = " + kv.getKey() + ", value = " + kv.getValue());
+        }
+    }
+
+    private void initTable() throws IOException {
+        Options options = new Options();
+        options.setString(METASTORE.key(), catalogType);
+        options.setString(URI.key(), metastoreUri);
+        options.setString(WAREHOUSE.key(), warehousePath);
+        Catalog catalog = CatalogFactory.createCatalog(CatalogContext.create(options));
+        Identifier identifier = new Identifier(databaseName, tableName);
+        try {
+            this.table = catalog.getTable(identifier);
+        } catch (Catalog.TableNotExistException e) {
+            LOG.error("Failed to open the paimon reader.", e);
+            throw new IOException("Failed to open the paimon reader.", e);
+        }
+    }
+
+    private void parseRequiredTypes() {
+        List<String> fieldNames = PaimonScannerUtils.fieldNames(table.rowType());
+        requiredTypes = new ColumnType[requiredFields.length];
+        logicalTypes = new DataType[requiredFields.length];
+        for (int i = 0; i < requiredFields.length; i++) {
+            int index = fieldNames.indexOf(requiredFields[i]);
+            if (index == -1) {
+                throw new RuntimeException(String.format("Cannot find field %s in schema %s",
+                        requiredFields[i], fieldNames));
+            }
+            DataType dataType = table.rowType().getTypeAt(index);
+            String type = PaimonTypeUtils.fromPaimonType(dataType);
+            requiredTypes[i] = new ColumnType(type);
+            logicalTypes[i] = dataType;
+        }
+    }
+
+    private void initReader() throws IOException {
+
+        ReadBuilder readBuilder = table.newReadBuilder();
+        RowType rowType = table.rowType();
+        List<String> fieldNames = PaimonScannerUtils.fieldNames(rowType);
+        if (requiredFields.length < fieldNames.size()) {
+            int[] projected = Arrays.stream(requiredFields).mapToInt(fieldNames::indexOf).toArray();
+            readBuilder.withProjection(projected);
+        }
+        // TODO(wangriyu): readBuilder.withFilter()
+        // TODO(wangriyu): table.copy() tableWithDynamicOptions()
+        Split split = PaimonScannerUtils.decodeStringToObject(splitInfo);
+        reader = readBuilder.newRead().createReader(split);
+    }
+
+    @Override
+    public void open() throws IOException {
+        try (ThreadContextClassLoader ignored = new ThreadContextClassLoader(classLoader)) {
+            initTable();
+            parseRequiredTypes();
+            initOffHeapTableWriter(requiredTypes, requiredFields, fetchSize);
```

Comment:
do you need to check `requiredTypes.size() == requiredFields.size()`

Reply:
no need, requiredTypes is derived from requiredFields